### ISSUE TITLE
Use arrays for ordered items

### ISSUE THREAD

**Issue by epage** [aobatact/clap-serde#24]
Currently subcommands and args are objects in the data model.  A lot of serialization sources consider the objects to have an undefined order but order of these is important.  Rather than relying on implementation details of the serializers (and tools mucking with the raw data), I'd recommend switching these to arrays like in the old clap YAML support

```yaml
name: myapp
version: "1.0"
author: Kevin K. <kbknapp@gmail.com>
about: Does awesome things
args:
    - config:
        short: c
        long: config
        value_name: FILE
        help: Sets a custom config file
        takes_value: true
    - INPUT:
        help: Sets the input file to use
        required: true
        index: 1
    - verbose:
        short: v
        multiple: true
        help: Sets the level of verbosity
subcommands:
    - test:
        about: controls testing features
        version: "1.3"
        author: Someone E. <someone_else@other.com>
        args:
            - debug:
                short: d
                help: print debug information
```

**Comment by epage** [comment]
Interesting, it looks like your YAML syntax already does this which is different than toml/json.

**Comment by aobatact** [comment]
Sorry for delay.
~Is this means that deserializer should treat Args (or subcommand) as `Vec<(Sting, Arg)>`  not `HashMap<String,Arg>`?~

Edit:
Oh this is about serializer.
Is this that the order should be specified when serializing arg and subcommand?

**Comment by epage** [comment]
This is about deserialization.  At a glance, I'm not seeing any `HashMap`s but effectively the data model should be `Vec<(String, Arg)>`.  Ordering of args and subcommands can be significant and we should respect that ordering and make it clear that we do.  Some source crates will preserve ordering but that is an implementation detail, usually the underlying formats (like json) are defined as unordered.

**Comment by aobatact** [comment]
Sorry for delay. 
Fixed in #32.
